pkg/cmd/join/exec.go

```diff
@@ -114,7 +119,39 @@ func (o *Options) run() error {
 		return err
 	}
 	output = append(output, out...)
-	fmt.Printf("Deploying klusterlet agent. Please wait a few minutes then log onto the hub cluster and run the following command:\n\n"+
+
+	timeout := int64(300)
+	gvr := schema.GroupVersionResource{Group: "operator.open-cluster-management.io", Version: "v1", Resource: "klusterlets"}
+	watch, err := dynamicClient.Resource(gvr).Watch(context.TODO(), metav1.ListOptions{TimeoutSeconds: &timeout})
+	if err != nil {
+		return err
+	}
+	defer watch.Stop()
+
+	fmt.Printf("Waiting for the management components ready...\n")
+
+WatchLoop:
+	for {
+		event, ok := <-watch.ResultChan()
+		if !ok { //The channel is closed by Kubernetes, thus, user should check the pod status manually
+			return fmt.Errorf("the registration agent is not ready yet, please check it manually")
+		}
+
+		var klusterlet *operatorv1.Klusterlet = new(operatorv1.Klusterlet)
+		data, err := json.Marshal(event.Object)
+		if err != nil {
+			continue
+		}
+		json.Unmarshal(data, &klusterlet)
+
+		for _, condition := range klusterlet.Status.Conditions {
+			if condition.Status == "True" && condition.Type == "Available" {
+				break WatchLoop
```

Comment:
why use a `WatchLoop ` here. I think you can just break?

I might start a gorountine here

```
go waitUntilContitionIsTrue(ctx)
<-ctx.Done()
```

Reply:
`break` only breaks from this loop `for _, condition := range klusterlet.Status.Conditions`.
gorountine looks graceful, let me update the code then.

Thanks for your time on the reviewing.